org.scala-ide.sdt.core.tests/src/org/scalaide/core/testsetup/Bdd.scala

```diff
@@ -0,0 +1,55 @@
+package org.scalaide.core.testsetup
+
+import java.io.File
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.jdt.core.IJavaProject
+import org.junit.Assert
+import org.scalaide.core.IScalaProject
+import org.scalaide.util.eclipse.EclipseUtils
+
+import SDTTestUtils.enableAutoBuild
+import SDTTestUtils.getErrorMessages
+import SDTTestUtils.sourceWorkspaceLoc
+import SDTTestUtils.workspace
+
+/**
+ * Behavioral Driven Development methods.
+ */
+trait Bdd {
+  import SDTTestUtils._
+
+  def givenCleanWorkspaceForProjects(projects: IProject*): Unit = {
+    workspace.build(IncrementalProjectBuilder.CLEAN_BUILD, new NullProgressMonitor)
+    projects.foreach { project =>
+      Assert.assertTrue(getErrorMessages(project).isEmpty)
+    }
+  }
+}
+
+object Bdd {
+  implicit def jProjectToIProject(project: IJavaProject): IProject = project.getProject
```

Comment:
I'm not a fan of such implicit conversions, they are not worth of the few saved characters. Please remove them, the code is easier to understand without them.